knowledge_graph.py

```diff
@@ -0,0 +1,110 @@
+    super().__init__(
+      model=AutoModelForSeq2SeqLM.from_pretrained(model),
+      tokenizer=AutoTokenizer.from_pretrained(model),
+      *args,
+      **kwargs)
+
+  def _extract_entity_relationships(self, text) -> List[Dict[str, Any]]:
+    triplets = []
+    relation = ""
+    subject = ""
+    relation = ""
+    object_ = ""
+    text = text.strip()
+    current = 'x'
+    for token in text.replace("<s>", "").replace("<pad>", "").replace("</s>", "").split():
+      if token == "<triplet>":
+        current = 't'
+        if relation != '':
+          triplets.append(
+            { "subject": subject.strip(), "relation": relation.strip(), "object": object_.strip() })
+          relation = ''
+        subject = ''
+      elif token == "<subj>":
+        current = 's'
+        if relation != '':
+          triplets.append(
+            { "subject": subject.strip(), "relation": relation.strip(), "object": object_.strip() })
+        object_ = ''
+      elif token == "<obj>":
+        current = 'o'
+        relation = ''
+      else:
+        if current == 't':
+          subject += ' ' + token
+        elif current == 's':
+          object_ += ' ' + token
+        elif current == 'o':
+          relation += ' ' + token
+    if subject != '' and relation != '' and object_ != '':
+      triplets.append(
+        { "subject": subject.strip(), 'relation': relation.strip(), 'object': object_.strip() })
+    return triplets
+
+  def postprocess(
+    self,
+    model_outputs,
+    return_type=ReturnType.TEXT,
+    clean_up_tokenization_spaces=False) -> List[Dict[str, Any]]:
+    """Post process function"""
+
+    records = []
+    for output_ids in model_outputs["output_ids"][0]:
+      if return_type == ReturnType.TEXT:
+        result = self.tokenizer.decode(
+          output_ids,
+          skip_special_tokens=False,
+          clean_up_tokenization_spaces=clean_up_tokenization_spaces)
+        results = self._extract_entity_relationships(result)
+        records += results
+    return records
+  
+
+knowledge_graph = KnowledgeGraph()
+
+txt = """
+The two principles of justice according to Rawls' theory are:
+
+1) First Principle: Each person has the same indefeasible claim to a fully adequate scheme of equal basic liberties, which scheme is compatible with the same scheme of liberties for all. Examples include freedom of thought, freedom of speech, and the right to vote.
+
+2) Second Principle: Social and economic inequalities in the distribution of primary goods (such as health, income, mobility, the social bases of self-respect, etc.) are to satisfy two conditions:
+   a) They are to be attached to offices and positions open to all under conditions of fair equality of opportunity.
+   b) They are to be to the greatest benefit of the least-advantaged members of society (the difference principle).
+"""
+
+entity_relations = knowledge_graph.inference(txt)
+print({"knowledge graph":entity_relations})
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code seems to be for testing purposes and should not be included in the production code. Consider removing it or moving it to a separate test file.

```diff
-97: knowledge_graph = KnowledgeGraph()
-98: 
-99: txt = """
-100: The two principles of justice according to Rawls' theory are:
-101: 
-102: 1) First Principle: Each person has the same indefeasible claim to a fully adequate scheme of equal basic liberties, which scheme is compatible with the same scheme of liberties for all. Examples include freedom of thought, freedom of speech, and the right to vote.
-103: 
-104: 2) Second Principle: Social and economic inequalities in the distribution of primary goods (such as health, income, mobility, the social bases of self-respect, etc.) are to satisfy two conditions:
-105:    a) They are to be attached to offices and positions open to all under conditions of fair equality of opportunity.
-106:    b) They are to be to the greatest benefit of the least-advantaged members of society (the difference principle).
-107: """
-108: 
-109: entity_relations = knowledge_graph.inference(txt)
-110: print({"knowledge graph":entity_relations})
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->